app.go

```diff
@@ -113,57 +100,120 @@ type App struct {
 	latestRoute *Route
 	// newCtxFunc
 	newCtxFunc func(app *App) CustomCtx
-	// custom binders
-	customBinders []CustomBinder
 	// TLS handler
 	tlsHandler *TLSHandler
 	// Mount fields
 	mountFields *mountFields
-	// Indicates if the value was explicitly configured
-	configured Config
+	// Route stack divided by HTTP methods
+	stack [][]*Route
+	// Route stack divided by HTTP methods and route prefixes
+	treeStack []map[string][]*Route
+	// custom binders
+	customBinders []CustomBinder
 	// customConstraints is a list of external constraints
 	customConstraints []CustomConstraint
 	// sendfiles stores configurations for handling ctx.SendFile operations
 	sendfiles []*sendFileStore
+	// App config
+	config Config
+	// Indicates if the value was explicitly configured
+	configured Config
 	// sendfilesMutex is a mutex used for sendfile operations
 	sendfilesMutex sync.RWMutex
+	mutex          sync.Mutex
+	// Amount of registered routes
+	routesCount uint32
+	// Amount of registered handlers
+	handlersCount uint32
+	// contains the information if the route stack has been changed to build the optimized tree
+	routesRefreshed bool
 }
 
 // Config is a struct holding the server settings.
 type Config struct {
```

Comment:
I think we don't need to align Config structures for better readability

Reply:
Makes sense, will revert this in a bit

Reply:
Done